pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -135,6 +135,120 @@ func maxOverTime(step *types.RangeVectorStepData, _ []types.ScalarData, _ types.
 	return maxSoFar, true, nil, nil
 }
 
+var TsOfMinOverTime = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction:         DropSeriesName,
+	StepFunc:                       tsOfMinOverTime,
+	NeedsSeriesNamesForAnnotations: true,
+}
+
+func tsOfMinOverTime(step *types.RangeVectorStepData, _ []types.ScalarData, _ types.QueryTimeRange, emitAnnotation types.EmitAnnotationFunc, _ *limiter.MemoryConsumptionTracker) (float64, bool, *histogram.FloatHistogram, error) {
+	cmpFunc := func(existing, next promql.FPoint) bool {
+		return next.F <= existing.F || math.IsNaN(existing.F)
+	}
+
+	return tsOverTime(cmpFunc, step, emitAnnotation)
+}
+
+var TsOfMaxOverTime = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction:         DropSeriesName,
+	StepFunc:                       tsOfMaxOverTime,
+	NeedsSeriesNamesForAnnotations: true,
+}
+
+func tsOfMaxOverTime(step *types.RangeVectorStepData, _ []types.ScalarData, _ types.QueryTimeRange, emitAnnotation types.EmitAnnotationFunc, _ *limiter.MemoryConsumptionTracker) (float64, bool, *histogram.FloatHistogram, error) {
+	cmpFunc := func(existing, next promql.FPoint) bool {
+		return next.F >= existing.F || math.IsNaN(existing.F)
+	}
+
+	return tsOverTime(cmpFunc, step, emitAnnotation)
+}
+
+func tsOverTime(compareFn func(promql.FPoint, promql.FPoint) bool, step *types.RangeVectorStepData, emitAnnotation types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
+	head, tail := step.Floats.UnsafePoints()
+
+	if len(head) == 0 && len(tail) == 0 {
+		return 0, false, nil, nil
+	}
+
+	if step.Histograms.Any() {
+		emitAnnotation(annotations.NewHistogramIgnoredInMixedRangeInfo)
+	}
```

Comment:
Could you please add a test for this scenario, where there are both floats and histograms in the range?